core/drivers/stm32_iwdg.c

```diff
@@ -364,14 +361,11 @@ static TEE_Result stm32_iwdg_probe(const void *fdt, int node,
 	struct stm32_iwdg_device *iwdg = NULL;
 	TEE_Result res = TEE_SUCCESS;
 
-	/*
-	 * Do not consider IWDG instances assigned to non-secure world.
-	 * For compatibility with legacy IWDG compatible nodes, consider nodes
-	 * with status="okay" and secure-status="disabled" as IWDG assigned to
-	 * non-secure world.
-	 */
-	if (!(fdt_get_status(fdt, node) & DT_STATUS_OK_SEC))
+	if (!IS_ENABLED(CFG_WDT_SM_HANDLER)) {
```

Comment:
I'd prefer to have a compile time restriction (static assert or rule in conf.mk) rather than a driver probing to do nothing. + we may face issue with firewall config when probing firewall bus without wanting to support this peripheral

Reply:
Ok, i'll update *plat-stm32mp\*/conf.mk*